README.md
# MagicBanana
Magic Banana Game 
Student ID: 214211857 
Shyam Lingam 
SIT305
Platform: Android
Link for github: https://github.com/ShyamLingam/MagicBanana

App Overview:
Zoku	 the	 baby	 monkey	 has	 just	 discovered	 that	 his	 mother	 is	 sick.	 He	 had	
previously	 heard	 that	 there	 are	 magic	 bananas	 located within	the forest.
The objective of this game is to collect as many banana's as possible. Bonus points are collected when magic bananas are collected.

Features:
Easy Mode (Begineers)
- This mode has been created for younger kids or first time players. As the speed is at a normal rate makes it easier to get higher score
Intermediate Mode ( Advanced Users)
- This mode has been created for advanced users whom want fast game play and enjoy a challenge
In Game Music
Support on Android Devices


# Henry comments 13/April
- Not enough commits + changelog items to pass at this frequency.
- I couldn't find your text-based data. Create a root folder "data/" and put all your JSON files in there.
- You're still missing a licenses.txt + changelog.md file.

# Henry comments 17/April
- Getting better. You still need a data/ directory :)

# Henry comments 27/April
- This file still needs compile instructions, and directory explanation so I know where to find stuff.
- Your changelog needs a lot of work
- Your data hasn't progressed enough. You need to get to this very quickly, as you can't start building your game until you can at least load the levels/world of your game. So work on getting loading working (from data files) asap.



CHANGELOG.MD
Changelog

All notable changes to this project will be documented in this file.
This is for Magic Banana Project.
Only one Author will be accessing this change log: Shyam Lingam

15-04-2018

Added:

- Configured intents and onclick listeners for the button in the main menu to navigate to other activity pages.
- Added Banana icon image to the activities
- Added additional activity and class pages. (About and Settings)

Changed:

- MenuMain activity has been modifed with new colours and button sizes.

Removed:

N/A

2-05-2018

Added:

- Added Main game page
- Configured intent to navigate to game page

Changed:



Removed:

- Removed Top Bar Settings

7-05-2018

Added:

- Added Monkey icon and movement feature to game page
- Added moving banana icons
- Added Score Count to count how many bananas

Changed:

N/A

Removed:

N/A

8-05-2018

Added:

N/A

Changed:

-Changed Monkey Icon
-Change main banana icon

Removed:

-Removed Background Map

10-05-2018

Added:

-Add new background to suit game play

Changed:

-Change the enemy icon, to a poison icon for loss of points

Removed:

-N/A

16-05-2018

Added:



Changed:

-Modying layout for clean look
-Changed colours to suit

Removed:

-Removed extra banana icon above the score list on the game play mode
-N/A

17-05-2018

Added:

Add second activity for hard level game play
Add new map dispay
Add background music to app

Changed:


Removed:

-Removed Old Map display

18-05-2018

Added:

Completed the intent for the new mode game play intermediate mode
Finalised README and License txt files for uploading

Changed:


Removed:

-Removed Old Map display


23-07-2018

Added:

Provided access to UNIT SIT313 Unit Chair access to GitHub : frankjiang2018

Re-using this application for Project 1 for SIT313

Changed:

Modified Git Files

Removed:




app/src/test/java/com/example/shyam/magicbanana/ExampleUnitTest.java
package com.example.shyam.magicbanana;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Example local unit test, which will execute on the development machine (host).
 *
 * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
 */
public class ExampleUnitTest {
    @Test
    public void addition_isCorrect() throws Exception {
        assertEquals(4, 2 + 2);
    }
}

app/src/main/java/com/example/shyam/magicbanana/about.java
package com.example.shyam.magicbanana;

import android.content.Intent;
import android.media.MediaPlayer;
import android.nfc.Tag;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;

/**
 * Created by Shyam on 15/4/18.
 */

public class about extends AppCompatActivity {


    ArrayList<String> numberlist = new ArrayList<>();
    private static final String TAG = "about";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.about);
        Log.d(TAG, "OnCreate: Starting.");
        get_json();
    }
        public void get_json()
    {
        String json;
        try
        {
            InputStream is = getAssets() .open("score.json");
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();

            json = new String(buffer, "UTF-8");
            JSONArray jsonArray = new JSONArray(json);

            for(int i = 0; i<jsonArray.length();i++)
            {
                JSONObject obj = jsonArray.getJSONObject(i);

                if (obj.getString("first_name").equals("Sean"))
                {
                    numberlist.add(obj.getString("hscore"));
                }
            }

            Toast.makeText(getApplicationContext(),numberlist.toString(),Toast.LENGTH_LONG) .show();

        } catch (IOException e)
        {
            e.printStackTrace();
        } catch (JSONException e) {
            e.printStackTrace();
        }


        Button btn6 = (Button) findViewById(R.id.button6);

        btn6.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d(TAG, "OnClick: clicked button6.");

                Intent intent = new Intent(about.this, MainMenu.class);
                startActivity(intent);
            }
        });
    }


}



app/src/main/java/com/example/shyam/magicbanana/game2.java
package com.example.shyam.magicbanana;

import android.content.Intent;
import android.graphics.Point;
import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.Display;
import android.view.MotionEvent;
import android.view.View;
import android.view.WindowManager;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.Timer;
import java.util.TimerTask;

/**
 * Created by Shyam on 29/4/18.
 */


public class game2 extends AppCompatActivity {

    private static final String TAG = "game2";

    private TextView scoreLabel;
    private TextView startLabel;
    private ImageView monkey;
    private ImageView banana;
    private ImageView poison;
    private ImageView magic;

    //Size of Monkey
    private int frameHeight;
    private int monkeySize;
    private int screenWidth;
    private int screenHeight;



    //Position of Monkey

    private int monkeyY;
    private int bananaX;
    private int bananaY;
    private int poisonX;
    private int poisonY;
    private int magicX;
    private int magicY;


    //score
    private int score = 0;



    //Int Class
    private Handler handler = new Handler();
    private Timer timer = new Timer();


    // Status Check

    private boolean action_flg = false;
    private boolean start_flg = false;





    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.game2);

        scoreLabel = (TextView) findViewById(R.id.scoreLabel);
        startLabel = (TextView) findViewById(R.id.startLabel);
        monkey = (ImageView) findViewById(R.id.monkey);
        banana = (ImageView) findViewById(R.id.banana);
        poison = (ImageView) findViewById(R.id.poison);
        magic = (ImageView) findViewById(R.id.magic);


        //Screen Size

        WindowManager wm = getWindowManager();
        Display disp = wm.getDefaultDisplay();
        Point size = new Point();
        disp.getSize(size);

        screenWidth = size.x;
        screenHeight = size.y;


        //move away from screen

        banana.setX(-80);
        banana.setY(-80);
        poison.setX(-80);
        poison.setY(-80);
        magic.setX(-80);
        magic.setY(-80);


        scoreLabel.setText("Score : 0");




    }

    public void changePos(){

        hitCheck();


        // banana

        bananaX -= 20;
        if (bananaX <0) {
            bananaX = screenWidth + 20;
            bananaY = (int) Math.floor(Math.random() * (frameHeight - banana.getHeight()));
        }
        banana.setX(bananaX);
        banana.setY(bananaY);

        //poison attack
        poisonX -= 25;
        if (poisonX < 0) {
            poisonX = screenWidth +10;
            poisonY = (int) Math.floor(Math.random() * (frameHeight - poison.getHeight()));
        }
        poison.setX(poisonX);
        poison.setY(poisonY);

        //Magic Banana for bonus points
        magicX -= 30;
        if (magicX <0) {
            magicX = screenWidth + 5000;
            magicY = (int) Math.floor(Math.random() * (frameHeight - magic.getHeight()));
        }
        magic.setX(magicX);
        magic.setY(magicY);

        //Moving the monkey
        if (action_flg == true){
            //Touching
            monkeyY -= 20;

        } else {
            //Release
            monkeyY += 20;
        }


        //monkey position

        if (monkeyY < 0) monkeyY = 0;

        if (monkeyY > frameHeight - monkeySize) monkeyY = frameHeight - monkeySize;

        monkey.setY(monkeyY);

        scoreLabel.setText("Score : " + score);

    }

    public void hitCheck() {

        // if the banana hits the monkey

        int bananaCenterX = bananaX + banana.getWidth() / 2;
        int bananaCenterY = bananaY + banana.getHeight() / 2;

        if (0 <= bananaCenterX && bananaCenterX <= monkeySize &&
                monkeyY <= bananaCenterY && bananaCenterY <= monkeyY + monkeySize) {

            score += 5;
            bananaX = -10;

        }

        //pink
        int magicCenterX = magicX + magic.getWidth() / 2;
        int magicCenterY = magicY + magic.getHeight() / 2;

        if (0 <= magicCenterX && magicCenterX <= monkeySize &&
                monkeyY <= magicCenterY && magicCenterY <= monkeyY + monkeySize) {

            score += 10;
            magicX = -10;


        }

        // Poison attack
        int poisonCenterX = poisonX + poison.getWidth() / 2;
        int poisonCenterY = poisonY + poison.getHeight() / 2;

        if (0 <= poisonCenterX && poisonCenterX <= monkeySize &&
                monkeyY <= poisonCenterY && poisonCenterY <= monkeyY + monkeySize) {

            // Stop Timer
            timer.cancel();



            //Show Result
            Intent intent = new Intent(getApplicationContext(), result.class);
            intent.putExtra("SCORE", score);
            startActivity(intent);

        }

    }
    public boolean onTouchEvent(MotionEvent me) {

        if (start_flg == false) {

            start_flg = true;

            FrameLayout frame = (FrameLayout) findViewById(R.id.frame);
            frameHeight = frame.getHeight();

            monkeyY = (int)monkey.getY();

            monkeySize = monkey.getHeight();


            startLabel.setVisibility(View.GONE);

            timer.schedule(new TimerTask() {
                @Override
                public void run() {
                    handler.post(new Runnable() {
                        @Override
                        public void run() {
                            changePos();
                        }
                    });
                }
            }, 0, 20);

        } else {
            if (me.getAction() == MotionEvent.ACTION_DOWN) {
                action_flg = true;

            } else if (me.getAction() == MotionEvent.ACTION_UP){
                action_flg = false;
        }


        }

        return true;
    }

}




app/src/main/java/com/example/shyam/magicbanana/result.java
package com.example.shyam.magicbanana;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class result extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);

        TextView scoreLabel = (TextView) findViewById(R.id.scoreLabel);
        TextView highScoreLabel = (TextView) findViewById(R.id.highScoreLabel);

        int score = getIntent().getIntExtra("SCORE", 0);
        scoreLabel.setText(score + "");

        SharedPreferences settings = getSharedPreferences("GAME_DATA", Context.MODE_PRIVATE);
        int highScore = settings.getInt("HIGH_SCORE", 0);

        if (score > highScore) {
            highScoreLabel.setText("High Score : " + score);

            // Update High Score

            SharedPreferences.Editor editor = settings.edit();
            editor.putInt("HIGH_SCORE", score);
            editor.commit();

        } else {
            highScoreLabel.setText("High Score: " + highScore);

        }

    }

    public void tryAgain(View view) {
        startActivity(new Intent(getApplicationContext(), startgame.class));

    }

}


app/src/main/java/com/example/shyam/magicbanana/MainMenu.java
package com.example.shyam.magicbanana;

import android.content.Intent;
import android.media.MediaPlayer;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainMenu extends AppCompatActivity {

    MediaPlayer mysong;
    Button son;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_menu);


        son = (Button) findViewById(R.id.soundon);

        mysong = MediaPlayer.create(MainMenu.this, R.raw.main);


        mysong.start();

        //Configuring Buttons

        Button btn15 = (Button) findViewById(R.id.button2);
        Button btn3 = (Button) findViewById(R.id.button3);
        Button btn4 = (Button) findViewById(R.id.button);


        //Intents for the buttons on how to changes to different class page.


        btn3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent int2= new Intent(MainMenu.this, about.class);
                startActivity(int2);
            }
        });
        btn4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent int3= new Intent(MainMenu.this, startgame.class);
                startActivity(int3);
            }
        });

        btn15.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent int4= new Intent(MainMenu.this, settings.class);
                startActivity(int4);
            }
        });

    }



    // This method closes the application
    public void clickexit(View v) {
        moveTaskToBack(true);
        mysong.release();
        android.os.Process.killProcess(android.os.Process.myPid());
        System.exit(1);
    }
}


app/src/main/java/com/example/shyam/magicbanana/startgame.java
package com.example.shyam.magicbanana;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class startgame extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_startgame);


    Button btn10 = (Button) findViewById(R.id.button10);
        Button btn11 = (Button) findViewById(R.id.button11);
        Button btn8 = (Button) findViewById(R.id.button8);

        btn10.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent int1 = new Intent(startgame.this, game1.class);
                startActivity(int1);
            }
        });

        btn8.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent int2 = new Intent(startgame.this, game2.class);
                startActivity(int2);
            }
        });

        btn11.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent int3 = new Intent(startgame.this, MainMenu.class);
                startActivity(int3);
            }
        });

    }

}






app/src/main/java/com/example/shyam/magicbanana/game1.java
package com.example.shyam.magicbanana;

import android.content.Intent;
import android.graphics.Point;
import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.Display;
import android.view.MotionEvent;
import android.view.View;
import android.view.WindowManager;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.Timer;
import java.util.TimerTask;

/**
 * Created by Shyam on 29/4/18.
 */


public class game1 extends AppCompatActivity {

    private static final String TAG = "game1";

    private TextView scoreLabel;
    private TextView startLabel;
    private ImageView monkey;
    private ImageView banana;
    private ImageView poison;
    private ImageView magic;

    //Size of Monkey
    private int frameHeight;
    private int monkeySize;
    private int screenWidth;
    private int screenHeight;



    //Position of Monkey

    private int monkeyY;
    private int bananaX;
    private int bananaY;
    private int poisonX;
    private int poisonY;
    private int magicX;
    private int magicY;


    //score
    private int score = 0;



    //Int Class
    private Handler handler = new Handler();
    private Timer timer = new Timer();


    // Status Check

    private boolean action_flg = false;
    private boolean start_flg = false;





    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.game1);

        scoreLabel = (TextView) findViewById(R.id.scoreLabel);
        startLabel = (TextView) findViewById(R.id.startLabel);
        monkey = (ImageView) findViewById(R.id.monkey);
        banana = (ImageView) findViewById(R.id.banana);
        poison = (ImageView) findViewById(R.id.poison);
        magic = (ImageView) findViewById(R.id.magic);


        //Screen Size

        WindowManager wm = getWindowManager();
        Display disp = wm.getDefaultDisplay();
        Point size = new Point();
        disp.getSize(size);

        screenWidth = size.x;
        screenHeight = size.y;


        //move away from screen

        banana.setX(-80);
        banana.setY(-80);
        poison.setX(-80);
        poison.setY(-80);
        magic.setX(-80);
        magic.setY(-80);


        scoreLabel.setText("Score : 0");




    }

    public void changePos(){

        hitCheck();


        // banana

        bananaX -= 12;
        if (bananaX <0) {
            bananaX = screenWidth + 20;
            bananaY = (int) Math.floor(Math.random() * (frameHeight - banana.getHeight()));
        }
        banana.setX(bananaX);
        banana.setY(bananaY);

        //poison attack
        poisonX -= 16;
        if (poisonX < 0) {
            poisonX = screenWidth +10;
            poisonY = (int) Math.floor(Math.random() * (frameHeight - poison.getHeight()));
        }
        poison.setX(poisonX);
        poison.setY(poisonY);

        //Magic Banana for bonus points
        magicX -= 20;
        if (magicX <0) {
            magicX = screenWidth + 5000;
            magicY = (int) Math.floor(Math.random() * (frameHeight - magic.getHeight()));
        }
        magic.setX(magicX);
        magic.setY(magicY);

        //Moving the monkey
        if (action_flg == true){
            //Touching
            monkeyY -= 20;

        } else {
            //Release
            monkeyY += 20;
        }


        //monkey position

        if (monkeyY < 0) monkeyY = 0;

        if (monkeyY > frameHeight - monkeySize) monkeyY = frameHeight - monkeySize;

        monkey.setY(monkeyY);

        scoreLabel.setText("Score : " + score);

    }

    public void hitCheck() {

        // if the banana hits the monkey

        int bananaCenterX = bananaX + banana.getWidth() / 2;
        int bananaCenterY = bananaY + banana.getHeight() / 2;

        if (0 <= bananaCenterX && bananaCenterX <= monkeySize &&
                monkeyY <= bananaCenterY && bananaCenterY <= monkeyY + monkeySize) {

            score += 5;
            bananaX = -10;

        }

        //pink
        int magicCenterX = magicX + magic.getWidth() / 2;
        int magicCenterY = magicY + magic.getHeight() / 2;

        if (0 <= magicCenterX && magicCenterX <= monkeySize &&
                monkeyY <= magicCenterY && magicCenterY <= monkeyY + monkeySize) {

            score += 10;
            magicX = -10;


        }

        // Poison attack
        int poisonCenterX = poisonX + poison.getWidth() / 2;
        int poisonCenterY = poisonY + poison.getHeight() / 2;

        if (0 <= poisonCenterX && poisonCenterX <= monkeySize &&
                monkeyY <= poisonCenterY && poisonCenterY <= monkeyY + monkeySize) {

            // Stop Timer
            timer.cancel();



            //Show Result
            Intent intent = new Intent(getApplicationContext(), result.class);
            intent.putExtra("SCORE", score);
            startActivity(intent);

        }

    }
    public boolean onTouchEvent(MotionEvent me) {

        if (start_flg == false) {

            start_flg = true;

            FrameLayout frame = (FrameLayout) findViewById(R.id.frame);
            frameHeight = frame.getHeight();

            monkeyY = (int)monkey.getY();

            monkeySize = monkey.getHeight();


            startLabel.setVisibility(View.GONE);

            timer.schedule(new TimerTask() {
                @Override
                public void run() {
                    handler.post(new Runnable() {
                        @Override
                        public void run() {
                            changePos();
                        }
                    });
                }
            }, 0, 20);

        } else {
            if (me.getAction() == MotionEvent.ACTION_DOWN) {
                action_flg = true;

            } else if (me.getAction() == MotionEvent.ACTION_UP){
                action_flg = false;
        }


        }

        return true;
    }

}


